### PR TITLE
Add `humility monorail` subcommand

### PR DESCRIPTION
**Pull request by mkeeter**
This is a dramatic refactoring of the previous `humility vsc7448` subcommand.

It now assumes the presence of a cooperative `monorail` task exposing an Idolatry API, rather than making raw SPI calls to the VSC7448.  Using these APIs, the `humility monorail` subcommand adds new functionality to dump port status, and show / reset counters.

Requires [hubris#662](https://github.com/oxidecomputer/hubris/pull/622)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -166,7 +167,7 @@ variable.
 - [humility readvar](#humility-readvar): read and display a specified Hubris variable
 - [humility registers](#humility-registers): print Hubris registers
 - [humility rencm](#humility-rencm): query Renesas 8A3400X ClockMatrix parts
-- [humility rendmp](#humility-rendmp): Renesas digitial muliphase controller operations
+- [humility rendmp](#humility-rendmp): Renesas digital muliphase controller operations
```
Comment (bcantrill) [inline README.md:170]: Thank you for fixing this! ;)

File: cmd/monorail/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,731 @@
...
+    )?;
+    match value {
+        Ok(v) => {
+            let value = if let Value::Base(Base::U32(v)) = v {
+                v
+            } else {
+                bail!("Got bad reflected value: expected U32, got {:?}", v);
+            };
+            println!("{} => 0x{:x}", reg, value);
+            if value == 0x88888888 {
```
Comment (bcantrill) [inline cmd/monorail/src/lib.rs:149]: Where does this value come from?

File: cmd/monorail/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,731 @@
...
+            assert_eq!(op.args.members.len(), 1); // Sanity-check!
+            let ret_size = hubris.typesize(op.ok)? as u32;
+
+            // Here we go!
+            ops.push(Op::Push32(op.task.task())); // Task id
+            ops.push(Op::Push16(op.code)); // opcode
+            ops.push(Op::Push(0)); // port (payload)
+            ops.push(Op::Push(NUM_PORTS as u8)); // comparison target (dummy)
+            ops.push(Op::Label(label));
+            {
```
Comment (bcantrill) [inline cmd/monorail/src/lib.rs:367]: Love this convention!

File: README.md [hunk 4]
```diff
@@ -869,6 +869,10 @@ DESC       LOW          HIGH          SIZE ATTR  ID TASK
 we can see from the `map` output has been sized to only 256 bytes.)
 
 
+### `humility monorail`
+
+No documentation yet for `humility monorail`; pull requests welcome!
```
Comment (bcantrill) [inline README.md:874]: At some point -- though perhaps not with this commit -- it would be great to get docs on `humliity monorail`...

**Review by bcantrill — APPROVED** [review]
Looks great!

File: cmd/monorail/src/lib.rs (hunk 2 above)
Comment (mkeeter) [reply cmd/monorail/src/lib.rs:154]: It's built into the VSC7448's SPI mode; added a comment to explain.

File: README.md (hunk 4 above)
Comment (mkeeter) [reply README.md]: Yeah, dropped some docs so that the CLI output isn't empty

**Pull request merged**